src/main/java/org/reactome/addlinks/dataretrieval/pharos/PharosTargetsDataRetriever.java

```diff
@@ -0,0 +1,43 @@
+package org.reactome.addlinks.dataretrieval.pharos;
+
+import java.io.FileWriter;
+import java.io.IOException;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class PharosTargetsDataRetriever extends PharosDataRetriever
+{
+	// NOTE: It might be tempting to add "\n" at the end of the "lines" in this string below, but that will result in a "Bad request" response
+	// from the GraphQL service. Clearly, queries cannot contain real line breaks.
+	private static final String TARGETS_QUERY = "{ \"query\": \"query {"
+			+ "    targets {"
+			+ "        count"
+			+ "        targets(top:"+PharosDataRetriever.TOP_AMOUNT_TOKEN+" skip:"+PharosDataRetriever.SKIP_AMOUNT_TOKEN+") {"
+			+ "            uniprot"
+			+ "        }"
+			+ "    }"
+			+ "}\" }";
+
+
+	/**
+	 * Implements the logic to process JSON from Pharos.
+	 */
+	protected int processJSONArray(FileWriter fw, JSONObject jsonObj) throws IOException
+	{
+		JSONArray targets = (((JSONObject)((JSONObject) jsonObj.get("data")).get("targets")).getJSONArray("targets"));
+		for (int i = 0; i < targets.length(); i++)
+		{
+			fw.write(((JSONObject)targets.get(i)).get("uniprot") + "\n");
```

Comment:
Is it worth checking to see if the JSONObject has a "uniprot" key or it's result is not null?  Otherwise a null value will be written to the FileWriter.

Reply:
The query sent to the server specifies "uniprot" so there should be a "uniprot" key.

Reply:
Okay, I was just thinking about the possibility if it's somehow is not a key.

Reply:
Here's what the data looks like:
```
{
  "data": {
    "targets": {
      "count": 20412,
      "targets": [
        {
          "uniprot": "A8MVA2"
        },
        {
          "uniprot": "A8MX34"
        },
...
```
If there is no "uniprot" then something has gone wrong.

Reply:
Okay I see.  I see you added a logger statement to handle it.  Thanks!